hardware/raspberrypi/bootmodes/msd.md

```diff
@@ -35,12 +47,31 @@ Check that the output `0x3020000a` is shown. If it is not, then the OTP bit has
 
 If you wish, you can remove the `program_usb_boot_mode` line from config.txt, so that if you put the SD card into another Raspberry Pi, it won't program USB host boot mode. Make sure there is no blank line at the end of config.txt.
 
-## Prepare the USB mass storage device
+You can now boot from an USB Mass Storage device in the same way as booting from SD, see the following paragraph for Raspberry Pi 3B+, CM3+.
+
+## Raspberry Pi 3B+, CM3+
+
+The Raspberry Pi 3B+ and CM3+ support USB Mass Storage boot out of the box. The settings specific to the previous versions of Raspberry Pi do not have to be executed.
+
+This is verified using the following steps, this is the same [procedure](../../../installation/installing-images/) as for SD cards.
+
+On Windows you may have to initialise a (clean) harddrive before it's visible and has a drive letter (which may be required to write the image).
+
+After preparing the storage device, connect the drive to the Raspberry Pi and power up the Pi (mind extra USB power usage from external drive).
+After five to ten seconds, the Raspberry Pi should begin booting and show the rainbow splash screen on an attached display. Make sure that you do not have an SD card inserted in the Pi, since if you do, it will boot from that first.
+
+## Unsupported devices
+
+Including Raspberry Pi 1, 2 (V1.1), Compute Module, Zero
+
+The boot code for USB is stored in the BCM2837 device only, so the Pi 1 A/B, Pi 2 B (v1.1), and Pi Zero will all require SD cards as they are based on the BCM2835 and BCM2836. This boot code is stored in ROM (except Pi 4B) which by definition cannot be changed.
 
-Copy a Raspbian or Raspbian Lite installation image directly onto your USB device, in the same way that you would for an SD card. To perform this step, follow the instructions [here](../../../installation/installing-images/README.md), remembering to select the drive that corresponds to your USB mass storage device. Make sure that you use a recent version of Raspbian/Raspbian Lite.
+Regarding the Raspberry Pi 4, the bootcode is stored in [EEPROM](../booteeprom.md) and can be updated. Support for mass storage boot will be added in a future update. In the meantime you can use the alternative below. 
```

Comment:
@timg236 Does 'special bootcode.bin-only boot mode' allow USB-booting on the Pi 4 ?

Reply:
The current Raspberry Pi 4 bootloader does not support USB mass storage boot. It's possible to have the root file system on USB because start.elf loads the kernel from /boot or the network and then the kernel can read the root file system itself.
That's normally the workaround that people refer to, so the OS is booting from from USB but the bootloader is not.

Reply:
Right. But does the Pi4 need start.elf, cmdline.txt, etc. on the SD-card too, or does it (also) support having _only_ bootcode.bin on the SD-card?

Reply:
Pi4 needs start.elf, fixup.dat, kernel, overlays and dtbs on the sd-card (or TFTP server)

bootcode.bin is not required because it's loaded from the EEPROM

Reply:
Thanks @timg236 !
@Paul-Ver This means that the "In the meantime you can use the alternative below." text isn't relevant for the Pi 4.

Reply:
Ah okay; I wasn't quite sure if it'd be working; but couldn't make up why not.
We may want to add the steps to do the above to boot the OS from USB (with bootloader from USB), but I'm not quite sure on these steps.

For now I've put the Pi 4 in a sub in the "unsupported devices"